code/mob/living.dm

```diff
@@ -1683,10 +1683,162 @@ var/global/icon/human_static_base_idiocy_bullshit_crap = icon('icons/mob/human.d
+	if (P.proj_data)  //ZeWaka: Fix for null.ks_ratio
+		damage = round((P.power*P.proj_data.ks_ratio), 1.0)
+		stun = round((P.power*(1.0-P.proj_data.ks_ratio)), 1.0)
+
+	var/rangedprot = get_ranged_protection() //will be 1 unless overridden
+	if (P.proj_data) //Wire: Fix for: Cannot read null.damage_type
+		switch(P.proj_data.damage_type)
+			if (D_KINETIC)
+				if (stun > 0)
+					src.remove_stamina(min(round(stun/rangedprot, 0.5) * 30, 125)) //thanks to the odd scaling i have to cap this.
+					src.stamina_stun()
+
+				src.TakeDamage("chest", (damage/rangedprot), 0, 0, DAMAGE_BLUNT)
+				if (stat==0) lastgasp()
+
+			if (D_PIERCING)
+				if (stun > 0)
+					src.remove_stamina(min(round(stun/rangedprot) * 30, 125)) //thanks to the odd scaling i have to cap this.
+					src.stamina_stun()
+
+				src.TakeDamage("chest", damage/max((rangedprot/3), 1), 0, 0, DAMAGE_STAB)
+				if (stat==0) lastgasp()
+
+			if (D_SLASHING)
+				if (stun > 0)
+					src.remove_stamina(min(round(stun/rangedprot) * 30, 125)) //thanks to the odd scaling i have to cap this.
+					src.stamina_stun()
+
+				if (rangedprot > 1)
+					src.TakeDamage("chest", (damage/rangedprot), 0, 0, DAMAGE_BLUNT)
+				else
+					src.TakeDamage("chest", (damage*2), 0, 0, DAMAGE_CUT)
+
+			if (D_ENERGY)
+				if (stun > 0)
+					src.do_disorient(clamp(stun*4, P.proj_data.power*(1-P.proj_data.ks_ratio)*2, stun+80), weakened = stun*2, stunned = stun*2, disorient = min(stun,  80), remove_stamina_below_zero = 0)
+					src.emote("twitch_v")// for the above, flooring stam based off the power of the datum is intentional
+
+				if (isalive(src)) lastgasp()
+
+				if (src.stuttering < stun)
+					src.stuttering = stun
+				src.TakeDamage("chest", 0, (damage/rangedprot), 0, DAMAGE_BURN)
+
+			if (D_BURNING)
+				if (stun > 0)
+					src.remove_stamina(min(round(stun/rangedprot) * 30, 125)) //thanks to the odd scaling i have to cap this.
+					src.stamina_stun()
+
+				if (src.is_heat_resistant())
+					// fire resistance should probably not let you get hurt by welders
+					src.visible_message("<span class='alert'><b>[src] seems unaffected by fire!</b></span>")
+					return 0
+				src.TakeDamage("chest", 0, (damage/rangedprot), 0, DAMAGE_BURN)
+				src.update_burning(damage/rangedprot)
+
+			if (D_RADIOACTIVE)
+				if (stun > 0)
+					src.remove_stamina(min(round(stun/rangedprot) * 30, 125)) //thanks to the odd scaling i have to cap this.
+					src.stamina_stun()
+
+				src.changeStatus("radiation", damage SECONDS)
+				if (src.add_stam_mod_regen("projectile", -5))
+					SPAWN_DBG(30 SECONDS)
+						src.remove_stam_mod_regen("projectile")
+
+			if (D_TOXIC)
+				if (stun > 0)
+					src.remove_stamina(min(round(stun/rangedprot) * 30, 125)) //thanks to the odd scaling i have to cap this.
+					src.stamina_stun()
+
+				if (P.proj_data.reagent_payload)
+					src.TakeDamage("chest", (damage/rangedprot), 0, 0, DAMAGE_STAB)
+					if (stat==0) lastgasp()
```

Comment:
isalive

Reply:
If they're unconscious they won't be speaking anyways 🤷